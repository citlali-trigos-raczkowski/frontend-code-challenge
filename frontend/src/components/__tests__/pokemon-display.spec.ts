import { PokemonDisplayMockProp } from "../../../tests/mocks/mock-props";
import { shallowMount, createLocalVue } from "@vue/test-utils";
import PokemonDisplay from "../pokemon-display.vue";
import VueRouter from "vue-router";

const localVue = createLocalVue();
localVue.use(VueRouter);

describe("components/pokemon-display.vue", () => {
  it("renders correctly", () => {
    const wrapper = shallowMount(PokemonDisplay, {
      propsData: PokemonDisplayMockProp,
      localVue,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  //   it("button opens up detail view", async () => {
  //     const wrapper = shallowMount(PokemonDisplay, {
  //       propsData: PokemonDisplayMockProps,
  //       localVue,
  //     });

  //     const aWrapper = wrapper.find("router-link-stub");
  //     expect(aWrapper.attributes().to).toBe(sampleEvolutionPokemon.name);
  //   });
});
