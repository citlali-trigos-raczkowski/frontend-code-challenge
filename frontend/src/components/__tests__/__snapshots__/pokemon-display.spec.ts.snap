// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pokemon-display.vue renders correctly 1`] = `
<div>
  <v-container-stub
    class="lighten-5"
    id="gallery-tiles"
    tag="div"
  >
    <pokemon-tile-stub
      data-cy="gallery-tiles"
      id="pokemon-tile"
      pokemon="[object Object]"
      togglefavorite="[Function]"
      triggerreload="[Function]"
    />
  </v-container-stub>
   
  <!---->
</div>
`;
